### PR TITLE
lv2: reduce closure

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
gtk2 is optional

Before:
```
$ nix path-info -Sh ./result
/nix/store/ysdp0p56qpnx9kd6lyyx6rgbwnz97wq6-lv2-1.18.2	 327.1M
```
After:
```
$ nix path-info -Sh ./result
/nix/store/q1rcs1dkj4zrrd0nd3qk3dnn5akh1b9g-lv2-1.18.2	 141.5M
```

related: https://github.com/NixOS/nixpkgs/issues/159612

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/audio/lv2/default.nix [hunk 1]
```diff
@@ -9,8 +9,8 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-TokfvHRMBYVb6136gugisUkX3Wbpj4K4Iw29HHqy4F4=";
   };
 
-  nativeBuildInputs = [ pkg-config wafHook ];
-  buildInputs = [ gtk2 libsndfile python3 ];
+  nativeBuildInputs = [ pkg-config wafHook python3 ];
```
[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/development/libraries/audio/lv2/default.nix:12]: oh, that's where that's coming from

[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/development/libraries/audio/lv2/default.nix]: that's really awkward for cross compilation

[the remark this answers is not in the document]
Comment (jtojnar) [reply pkgs/development/libraries/audio/lv2/default.nix]: Maybe we could move the file to a `dev` output? Not really sure about what it is used for.

[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/development/libraries/audio/lv2/default.nix]: done

[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/development/libraries/audio/lv2/default.nix]: also, dev output fixes a lot of issues, including the closure propagation

File: pkgs/development/libraries/audio/lv2/default.nix [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
+
+# test derivations
+, pipewire
+}:
 
 stdenv.mkDerivation rec {
   pname = "lv2";
   version = "1.18.2";
 
+  outputs = [ "out" "dev" ];
```
Comment (jtojnar) [inline pkgs/development/libraries/audio/lv2/default.nix:11]: Probably also need `outputBin = "dev";`.

File: pkgs/applications/audio/magnetophonDSP/LazyLimiter/default.nix [hunk 3]
```diff
@@ -10,7 +11,7 @@ stdenv.mkDerivation rec {
     sha256 = "10xdydwmsnkx8hzsm74pa546yahp29wifydbc48yywv3sfj5anm7";
   };
 
-  buildInputs = [ faust2jaqt faust2lv2 ];
+  buildInputs = [ faust2jaqt faust2lv2 lv2 ];
```
Comment (jtojnar) [inline pkgs/applications/audio/magnetophonDSP/LazyLimiter/default.nix:14]: Right, this should be fixed by correcting the symlinks in `lv2.dev` to point to `lv2.out`.

**Comment by jonringer** [comment]
honestly, I lost all will to continue this, this package is just a nebulous web of symlinks and hacks.

**Comment by jtojnar** [comment]
Looking at the code, it should work if we pass the correct `includedir` flag to waf:

https://github.com/lv2/lv2/blob/6cefc7df1a6158c79d23029df183c09b10b88cad/wscript#L347-L355

**Comment by jtojnar** [comment]
Yeah, the relativization is ugly but seems to work:

```patch
--- a/pkgs/development/libraries/audio/lv2/default.nix
+++ b/pkgs/development/libraries/audio/lv2/default.nix
@@ -18,7 +18,11 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ pkg-config wafHook ];
   buildInputs = [ libsndfile python3 ];
 
-  wafConfigureFlags = lib.optionals stdenv.isDarwin [ "--lv2dir=${placeholder "out"}/lib/lv2" ];
+  wafConfigureFlags = [
+    "--includedir=${placeholder "dev"}/include"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "--lv2dir=${placeholder "out"}/lib/lv2"
+  ];
 
   passthru.tests = {
     inherit pipewire;
```

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 159875` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>audacity</li>
    <li>rkrlv2</li>
    <li>suil-qt5</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>audacity-gtk2 (audacity-gtk3)</li>
    <li>gxmatcheq-lv2</li>
    <li>gxplugins-lv2</li>
    <li>suil (suil-qt4)</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>lilv</li>
    <li>lv2</li>
    <li>lvtk</li>
    <li>sratom</li>
  </ul>
</details>

File: pkgs/development/libraries/audio/lv2/default.nix [hunk 4]
```diff
@@ -1,18 +1,32 @@
-{ lib, stdenv, fetchurl, gtk2, libsndfile, pkg-config, python3, wafHook }:
```
Comment (ajs124) [inline pkgs/development/libraries/audio/lv2/default.nix:1]: If we're dropping `gtk2` from `buildInputs`, shouldn't we also drop it here?

Also, this probably supersedes #151292 then (cc @cheriimoya).

Comment (jonringer) [reply pkgs/development/libraries/audio/lv2/default.nix:1]: I'm not dropping it anymore.

Exclusion of gtk2 also makes generation of some extensions not possible.

Comment (jonringer) [reply pkgs/development/libraries/audio/lv2/default.nix:1]: oh, looks like I forgot to push some changes.. I'll be back on my desktop tomorrow.

Comment (jtojnar) [reply pkgs/development/libraries/audio/lv2/default.nix:1]: I think at this point it is preferable to drop `gtk2`. If someone uses the `scope` example plug-in for tests like [Carla](https://github.com/falkTX/Carla/blob/b8d38e395c215eed189a4849972f90af810ef971/source/tests.old/CarlaUtils2.cpp#L88) did, they can easily replace it with another plug-in. If they really want to use that plug-in, they can build a separate derivation. I would not expect anyone to actually need that and getting rid of GTK 2 on the ISO would be nice.

Comment (cheriimoya) [reply pkgs/development/libraries/audio/lv2/default.nix:1]: > I think at this point it is preferable to drop `gtk2`.

Exactly what i thought when opening #151292

Comment (jtojnar) [reply pkgs/development/libraries/audio/lv2/default.nix:1]: Amended the commit to drop it here as well.

**Review by jtojnar — APPROVED** [review]

**Pull request merged**